src/test/java/com/spotify/google/cloud/pubsub/client/PublisherTest.java

```diff
@@ -110,21 +117,82 @@ public void testOutstandingRequests() throws InterruptedException, ExecutionExce
 
     // Publish a message and verify that the outstanding request counter rises to 1
     final CompletableFuture<String> f1 = publisher.publish("t1", m1);
-    assertThat(publisher.outstandingRequests(), is(1));
+    assertThatWithin(5, SECONDS, publisher::outstandingRequests, is(1));
```

Comment:
Maybe you could have avoided having your tests depend on time, if you would mock the ScheduledExecutorService and then control it when it gets fired. 

I think that would make tests less flaky and more precise.